core/src/main/java/com/linecorp/armeria/server/annotation/AnnotatedHttpDocServicePlugin.java

```diff
@@ -0,0 +1,417 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.annotation;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static com.linecorp.armeria.server.PathMappingPrefixes.PREFIX;
+import static com.linecorp.armeria.server.PathMappingPrefixes.REGEX;
+import static com.linecorp.armeria.server.annotation.AnnotatedHttpDocServiceUtil.getNormalizedTriePath;
+import static java.util.Objects.requireNonNull;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.HttpHeaderPathMapping;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceConfig;
+import com.linecorp.armeria.server.docs.DocServicePlugin;
+import com.linecorp.armeria.server.docs.EndpointInfo;
+import com.linecorp.armeria.server.docs.EndpointInfoBuilder;
+import com.linecorp.armeria.server.docs.EnumInfo;
+import com.linecorp.armeria.server.docs.ExceptionInfo;
+import com.linecorp.armeria.server.docs.FieldInfo;
+import com.linecorp.armeria.server.docs.FieldRequirement;
+import com.linecorp.armeria.server.docs.MethodInfo;
+import com.linecorp.armeria.server.docs.NamedTypeInfo;
+import com.linecorp.armeria.server.docs.ServiceInfo;
+import com.linecorp.armeria.server.docs.ServiceSpecification;
+import com.linecorp.armeria.server.docs.StructInfo;
+import com.linecorp.armeria.server.docs.TypeSignature;
+
+/**
+ * A {@link DocServicePlugin} implementation that supports the {@link AnnotatedHttpService}.
+ */
+public class AnnotatedHttpDocServicePlugin implements DocServicePlugin {
+
+    @VisibleForTesting
+    static final TypeSignature VOID = TypeSignature.ofBase("void");
+    @VisibleForTesting
+    static final TypeSignature BOOL = TypeSignature.ofBase("boolean");
+    @VisibleForTesting
+    static final TypeSignature INT8 = TypeSignature.ofBase("int8");
```

Comment:
I borrowed them from the [OpenAPI specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#data-types)
so that we can use these types in `DocService` and `OpenAPI`. Do you think it's  OK? or should I use the Java type names?

Reply:
Ah, OK, then. Let's leave them as they are.